### ISSUE TITLE
RealSense error calling rs2_get_frame_metadata (frame:0x55bb489ba208, frame_metadata:Actual Exposure)

### ISSUE THREAD

**Issue by sandeshk1** [WKDSMRT/realsense-gstreamer#19]
RealSense Camera Model: D435i

$ gst-launch-1.0 realsensesrc ! videoconvert ! autovideosink
Setting pipeline to PAUSED ...
Pipeline is live and does not need PREROLL ...
Setting pipeline to PLAYING ...
New clock: GstSystemClock
ERROR: from element /GstPipeline:pipeline0/GstRealsenseSrc:realsensesrc0: RealSense error calling rs2_get_frame_metadata (frame:0x55a4117a2cf8, frame_metadata:Actual Exposure)
Additional debug info:
../src/gstrealsensesrc.cpp(348): gst_realsense_src_create (): /GstPipeline:pipeline0/GstRealsenseSrc:realsensesrc0
Execution ended after 0:00:00.003126435
Setting pipeline to PAUSED ...
Setting pipeline to READY ...
Setting pipeline to NULL ...
Freeing pipeline ...

**Comment by TimConnelly** [comment]
Hi, can you try this branch? https://github.com/WKDSMRT/realsense-gstreamer/tree/bugfix-pipeline-start-error

Getting the actual exposure is set to 0 on that branch to avoid the error. I'll see if I can find a fix.

**Comment by sandeshk1** [comment]
Hi, 

Thanks for the quick reply.

I am facing the same error in the  bugfix-pipeline-state-error branch as well.

Is there any way I can unblock from this issue.

**Comment by sandeshk1** [comment]
Hi, sry there was an error while pulling the bugfix branch. I see that you have commented the part where actual exposure is read from realsense camera to unblock this issue. So bugfix branch is unblocking this issue.

Can you please try to provide a fix for this

**Comment by TimConnelly** [comment]
Yes. I'll look into it. Do you need the exposure value for your application?

**Comment by sandeshk1** [comment]
Thanks for looking into it.
It would be great to have a property to set the exposure value for the real sense camera .

**Comment by TimConnelly** [comment]
@sandeshk1 Can you pull and try the [bugfix-pipeline-start-error](https://github.com/WKDSMRT/realsense-gstreamer/tree/bugfix-pipeline-start-error) branch? Getting the exposure is working for me now. If it works for you then I'll move on to looking at adding an exposure property.

**Comment by sandeshk1** [comment]
I am still getting the same error in the https://github.com/WKDSMRT/realsense-gstreamer/tree/bugfix-pipeline-start-error branch

**Comment by TimConnelly** [comment]
Can you check which version of librealsense2 you're using with `apt list | grep realsense`? The error stopped occurring on my end when I installed the latest version of librealsense2 (2.41.0). 

> librealsense2-dev/focal,now 2.41.0-0~realsense0.3963 amd64 [installed]

**Comment by PCH10507323** [comment]
Hi, follow the previous commits, I build the source with this branch

> Hi, can you try this branch? https://github.com/WKDSMRT/realsense-gstreamer/tree/bugfix-pipeline-start-error
> 
> Getting the actual exposure is set to 0 on that branch to avoid the error. I'll see if I can find a fix.

Also, I installed libgstreamer-plugins-base1.0-dev and my librealsense2 version is 2.41.0
But I still have this error
```
RealSense error calling rs2_get_frame_metadata (frame:0x55a4117a2cf8, frame_metadata:Actual Exposure)
```
Is there any part I miss to setup? Thanks

**Comment by sandeshk1** [comment]
@TimConnelly I had installed realsense v2.41.0 from source(https://github.com/IntelRealSense/librealsense) and still faced the exposure issue. 

I had to comment the line where exposure metadata is being read to get it working

**Comment by TimConnelly** [comment]
@sandeshk1 I'm no longer able to reproduce the error on my end. Can you send the output of `gst-inspect-1.0 realsensesrc` and the pipeline error. 

Also, which specific OS, including version, are you developing on?

**Comment by TimConnelly** [comment]
@sandeshk1 Are you still experiencing this problem?

I was able to replicate the error on my old system that had version 2.39.0 of librealsense. I upgraded to version 2.41.0 (following the [instructions here](https://github.com/IntelRealSense/librealsense/blob/master/doc/distribution_linux.md)) and the error resolved.  

When you installed from source, which options did you give to Cmake? From the comment on [this realsense issue](https://github.com/IntelRealSense/librealsense/issues/6689#issuecomment-649240755) the FORCE_RSUSB_BACKEND option can affect performance.

**Comment by mcaniot** [comment]
I have the same issue using realsense version 2.50. I already tried the bug-fixed branch without any success.
